### DISCUSSION TITLE
Documentation: Host

### DISCUSSION THREAD

**Discussion by ShalokShalom** [fastfetch-cli/fastfetch#347] [Q&A] [answered]
# Current state:

I have no idea where fastfetch sources its information about Host:

# Wanted state:

I like to know how to change the source of Host. 

# Why the change is sensible:

I installed my OS on an enclosed SSD and use that on several different devices. 
As I installed it, I did so on an old Macbook Air, and now I use it on a desktop computer. 

![image](https://user-images.githubusercontent.com/6344099/184823492-73b1e12a-1fe1-4c21-9c9c-4c1cd1043824.png)

"Host" also suggests hostname, which it isnt. I like to change the output of "Host:" and dont know, where fastfetch gets this info from. 

Thanks

**Comment by LinusDierheimer** [comment, marked as answer]
TLDR: [the code](https://github.com/LinusDierheimer/fastfetch/blob/master/src/modules/host.c)

Basically it reads files in `/sys/devices/virtual/dmi/id/` and falls back to files in `/sys/class/dmi/id/`. You are probably looking for the files `product_name` and `product_version`. Those are values provided by your hardware, and just made accessible through the kernel. Changing them might be impossible for some values. If you manage to do it, remember to run with `--recache` once.

If you want to change the fastfetch output, use the following flags:
```bash
fastfetch --host-key "Hardware" --host-value "MacBook Air"
```

To do this permanently, put those flags in `~/.config/fastfetch.conf`.
To have a more flexible format string, look at `fastfetch --help host-format`

**Reply by ell1e** [reply]
How would one put those flags into `~/.config/fastfetch/config.jsonc`?

**Comment by ShalokShalom** [comment]
>  Those are values provided by your hardware, and just made accessible through the kernel. 

Well, the point is, it is reporting the value of the hardware, on which I originally installed it. 

I am now on the desktop, and it says Macbook Air. 

Thanks for your help
